### DISCUSSION TITLE
Difference between "updated_at" and "pushed_at" in repositories list response

### DISCUSSION THREAD

**Discussion by bartosz-kuros** [community/community#24442] [Apps, API and Webhooks]
<p>Hi, can I get more detailed info on what’s the difference between these two fields in </p>
<pre><code>GET /user/repos

...  
" **updated\_at**": "2019-04-16T14:18:39Z",  
" **pushed\_at**": "2019-04-16T14:18:38Z",  
...
</code></pre>
<p>How do I push to a repo without updating it as I’ve seen repositories with “updated_at” earlier than “pushed_at”?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/docc46">@docc46</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<blockquote>
<p>Hi, can I get more detailed info on what’s the difference between these two fields in</p>
</blockquote>
<p>pushed_at will be updated any time a commit is pushed to any of the repository’s branches.</p>
<p>updated_at will be updated any time the repository object is updated, e.g. when the description or the primary language of the repository is updated. The difference is that pushed_at represents the date and time of the last commit, whereas the updated_at represents the date and time of the last change the the repository. A change to the repository might be a commit, but it may also be other things, such as changing the description of the repo, creating wiki pages, etc. In other words, commits are a subset of updates, and the pushed_at timestamp will therefore either be the same as the updated_at timestamp, or it will be an earlier timestamp. (<a href="https://stackoverflow.com/questions/15918588/github-api-v3-what-is-the-difference-between-pushed-at-and-updated-at" rel="nofollow noopener">stackoverflow</a>)</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Reply by zoe-y-zuo** [reply]
Hi thanks for the answer! 

So the pushed_at timestamp should be either the same as the updated_at timestamp, or earlier, right? But I'm looking at an updated_at timestamp earlier than the same repo's pushed_at...    
pushed: 2022-08, updated: 2022-07. 

Why is this?

**Reply by clayne** [reply]
No, a repository can be setup with description, etc. and left alone and then commits are pushed to the _git_ repository involved with it continuously after that point.

`updatedAt` involves the GH-specific set of attributes associated with the "repo."
`pushedAt` involves the time of the last git commit to the git repo directly involved with the GH "repo."

I use repo in quotes here as GH's concept of a repo is a superset containing an actual git repo itself.
